### DISCUSSION TITLE
(PVE Perl Style cont.) Formatting multi-line conditional expression

### DISCUSSION THREAD

**Discussion by cheese** [perltidy/perltidy#100] [Q&A]
perltidyrc is the same in https://github.com/perltidy/perltidy/discussions/96

What PVE prefers:
```
if (
    !defined($another_variable_name)
    && defined($some_hash->{value_foo})
    && $another_condition
    && defined($yet_another_check)
) {
    $a_longer_variable_name = $some_hash->{value_foo};
}
```

What I got from perltidy:
```
if (  !defined($another_variable_name)
    && defined($some_hash->{ value_foo })
    && $another_condition
    && defined($yet_another_check)) {
    $a_longer_variable_name = $some_hash->{ value_foo };
}
```

Most notably, the opening parenthesis is placed at the end of line and the closing one at the start of line.

**Comment by shancock9** [comment]
It would be possible to add a flag to perltidy to break the parens open here to give the desired format.  In fact it already does this if there is a blank line or side comment anywhere within the parens; for example:
```
if (
      !defined($another_variable_name)
    && defined($some_hash->{ value_foo })

    && $another_condition
    && defined($yet_another_check)
) {
    $a_longer_variable_name = $some_hash->{ value_foo };
}
```
A flag could be added to make this happen even without a blank line.
